packages/jest-jasmine2/src/error_on_private.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Global} from '../../../types/Global';
+
+// prettier-ignore
+const disabledGlobals = {
+  fail: 'Illegal usage of global `fail`, prefer throwing an error, or the `done.fail` callback.',
+  pending: 'Illegal usage of global `pending`, prefer explicitly skipping a test using `test.skip`',
+  spyOn: 'Illegal usage of global `spyOn`, prefer `jest.spyOn`.',
+  spyOnProperty: 'Illegal usage of global `spyOnProperty`, prefer `jest.spyOn`.',
+};
+
+// prettier-ignore
+const disabledJasmineMethods = {
+  addMatchers: 'Illegal usage of `jasmine.addMatchers`, prefer `expect.extends`.',
+  any: 'Illegal usage of `jasmine.any`, prefer `expect.any`.',
+  anything: 'Illegal usage of `jasmine.anything`, prefer `expect.anything`.',
+  arrayContaining: 'Illegal usage of `jasmine.arrayContaining`, prefer `expect.arrayContaining`.',
+  createSpy: 'Illegal usage of `jasmine.createSpy`, prefer `jest.fn`.',
+  objectContaining: 'Illegal usage of `jasmine.objectContaining`, prefer `expect.objectContaining`.',
+  stringMatching: 'Illegal usage of `jasmine.stringMatching`, prefer `expect.stringMatching`.',
+};
+
+export function installErrorOnPrivate(global: Global): void {
+  const {jasmine} = global;
+  Object.keys(disabledGlobals).forEach(functionName => {
+    global[functionName] = () => {
+      throwAtFunction(disabledGlobals[functionName], global[functionName]);
+    };
+  });
+
+  Object.keys(disabledJasmineMethods).forEach(methodName => {
+    jasmine[methodName] = () => {
+      throwAtFunction(disabledJasmineMethods[methodName], jasmine[methodName]);
+    };
+  });
+
+  const original = jasmine.DEFAULE_TIMEOUT_INTERVAL;
+  // $FlowFixMe Flow seems to be confused about accessors and tries to enfoce having a `value` property.
+  Object.defineProperty(jasmine, 'DEFAULT_TIMEOUT_INTERVAL', {
+    configurable: true,
+    enumerable: true,
+    get: () => original,
+    set: () => {
+      throwAtFunction(
+        'Illegal usage of `jasmine.DEFAULT_TIMEOUT_INTERVAL`, prefer `jest.setTimeout`.',
+        jasmine.set,
+      );
```

Comment:
Whoops. Jest actually sets this under the hood, so I'll have to find a different way to do this.